File: lab21.cs
using System;

class StringClass
{
    protected string value;

    // Конструктор з параметром
    public StringClass(string str)
    {
        value = str;
    }

    // Метод для обчислення довжини рядка
    public int Length()
    {
        return value.Length;
    }

    // Метод для виведення рядка
    public void Print()
    {
        Console.WriteLine("Рядок: " + value);
    }
}

class DigitString : StringClass
{
    // Конструктор з параметром
    public DigitString(string str) : base(str) { }

    // Метод для видалення заданого символу
    public void RemoveChar(char c)
    {
        value = value.Replace(c.ToString(), "");
    }

    // Метод отримання даних об’єкту
    public string GetValue()
    {
        return value;
    }
}

class Program
{
    static void Main()
    {
        // Створення об'єкта похідного класу
        DigitString ds = new DigitString("12345");
        ds.Print();

        // Обчислення і виведення довжини
        Console.WriteLine("Довжина рядка: " + ds.Length());

        // Видалення символу '3' і повторний вивід
        ds.RemoveChar('3');
        ds.Print();
        Console.WriteLine("Довжина рядка після видалення: " + ds.Length());
    }
}

File: lab21.cpp
#include <iostream>
#include <string>

class String {
protected:
    std::string value;

public:
    // Конструктор з параметром
    String(const std::string& str) : value(str) {}

    // Метод для обчислення довжини рядка
    int length() const {
        return value.length();
    }

    // Метод для виведення рядка
    void print() const {
        std::cout << "Рядок: " << value << std::endl;
    }
};

class DigitString : public String {
public:
    // Конструктор з параметром
    DigitString(const std::string& str) : String(str) {}

    // Метод для видалення заданого символу
    void removeChar(char c) {
        value.erase(std::remove(value.begin(), value.end(), c), value.end());
    }

    // Метод отримання даних об’єкту
    std::string getValue() const {
        return value;
    }
};

int main() {
    // Створення об'єкта похідного класу
    DigitString ds("12345");
    ds.print();
    
    // Обчислення і виведення довжини
    std::cout << "Довжина рядка: " << ds.length() << std::endl;
    
    // Видалення символу '3' і повторний вивід
    ds.removeChar('3');
    ds.print();
    std::cout << "Довжина рядка після видалення: " << ds.length() << std::endl;

    return 0;
}
